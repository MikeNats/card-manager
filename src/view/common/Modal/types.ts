export type ModalType = {
  isVisible: boolean, 
  setVisibility: React.EventHandler<any>, 
  children?: React.ReactNode,
  title: string
}