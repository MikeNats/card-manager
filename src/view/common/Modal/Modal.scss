.comp-modal{
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    z-index: 99;
    overflow: hidden;
    display: flex;
    justify-content: center;
    transition: height 1s ease 3s;
    .bg{
        @include background-opacity($grey-20, 0.8); 
        position: fixed;
        z-index: 90;
        top:0;    
        left:0;
        width:100%;
        height:100%;
    } 
    .icon{ 
        background-image:url('/exit.png');
        width:16px;
        height: 14px;
        position: absolute;
        right: 27px;
        top:30px;
        cursor: pointer;
    }
    .content{
        width: 100%;
        background: #FFFFFF;
        border-radius: 16px;
        padding: 62px 32px 0 32px;
        display: flex;
        flex-direction: column;
        max-width: 600px;
        position: relative;
        z-index: 99;
        margin-top: 70px;
        transition: all 0.4s ease;
    }

    h2{
        color: #000;
        margin: 0 0 30px 0;
        font-size: 20px;
        font-weight: bold;
    } 
}
.comp-modal.hide{
    height: 0;
    z-index: -1;
    .content{
        margin-top: 2000px;
        height: 0; 
    }   
    .bg{
        display: none;
    }
}
