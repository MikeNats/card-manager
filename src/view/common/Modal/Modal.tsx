
import React from 'react'
import {ModalType} from './types'
const Modal = ({isVisible=false, setVisibility, title, children}:ModalType) => {
 const hideModal = () => setVisibility(false)
 return <div className={`comp-modal ${isVisible ? '' : 'hide'}`}>
   <div className="bg"></div>
    <div className="content">
      <span onClick={hideModal} className="icon"></span>
      <h2>{title}</h2>
      {children} 
    </div>
  </div>} 

export default React.memo(Modal); 