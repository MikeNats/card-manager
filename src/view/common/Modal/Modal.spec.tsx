import React from 'react'
import Modal from './Modal'
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() })

describe("Modal", () => {
    let onChangeHandler = jest.fn()
    
    it('renders as expected', () => {
     const  component = shallow(<Modal
        isVisible={false}
        title="test"
        setVisibility={onChangeHandler}
        ><p>test</p></Modal>)

      expect(component).toMatchSnapshot();
    })

      it('should set props', () => {	
        const component = shallow(<Modal  
          isVisible={false}
          title="test"
          setVisibility={onChangeHandler}
          ><p>test</p></Modal>)
          expect(component.props().className).toEqual('comp-modal hide');
          expect(component.find('h2').text()).toEqual('test')
      });

      it('should populate given children', () => {	
        const component = mount(<Modal  
          isVisible={false}
          title="test"
          setVisibility={onChangeHandler}
          ><p>test</p></Modal>).find('p');
          expect(component).toBeTruthy();
          expect(component.text()).toEqual('test')
      });

      it('should invoke `clickHandler` on click', () => {
          const event = { persist: ()=>{}, currentTarget: {value: "spam"}};	
          mount(<Modal  
            isVisible={false}
            title="test"
            setVisibility={onChangeHandler}
            ><p>test</p></Modal>).find('span').simulate('click', event)
          
          expect(onChangeHandler.mock.calls.length).toEqual(1);
      });
});
