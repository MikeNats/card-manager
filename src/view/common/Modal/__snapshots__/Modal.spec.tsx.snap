// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders as expected 1`] = `ShallowWrapper {}`;
